src/applications/vaos/new-appointment/redux/helpers/formSubmitTransformers.v2.js

```diff
@@ -155,12 +155,23 @@ export function transformFormToVAOSVARequest(state) {
   };
 }
 
-function getUserMessage(data) {
-  const label = PURPOSE_TEXT.find(
+// function getUserMessage(data) {
+//   const label = PURPOSE_TEXT.find(
+//     purpose => purpose.id === data.reasonForAppointment,
+//   ).short;
+
+//   return `${label}: ${data.reasonAdditionalInfo}`;
+// }
+
+function getReasonCode(data) {
+  const code = PURPOSE_TEXT.filter(purpose => purpose.id !== 'other').find(
     purpose => purpose.id === data.reasonForAppointment,
-  ).short;
+  )?.short;
 
-  return `${label}: ${data.reasonAdditionalInfo}`;
+  return {
+    coding: code ? [{ code }] : [],
+    text: data.reasonAdditionalInfo,
```

Comment:
we return `text` key value pair only if user selects "My reason isn't listed". Plus, the `text` field may be limited to 100 characters. Not quite sure how to code it in ternary expression or maybe a simple if statement will be better?
```suggestion
    text: !code? data.reasonAdditionalInfo: null,
```